Frameworks/Foundation/NSSet.mm

```diff
@@ -388,24 +394,39 @@ - (void)enumerateObjectsUsingBlock:(void (^)(id, BOOL*))block {
 }
 
 /**
- @Status Caveat
- @Notes options parameter not supported
+ @Status Interoperable
+ @Notes NSEnumerationReverse is undefined on reference platform so we will ignore it
 */
 - (void)enumerateObjectsWithOptions:(NSEnumerationOptions)options usingBlock:(void (^)(id, BOOL*))block {
-    if (options & NSEnumerationReverse) {
-        assert(0);
-    } else {
-        BOOL stop = FALSE;
-        int i = 0;
+    dispatch_queue_t queue;
+    dispatch_group_t group;
 
-        for (id curObj in self) {
+    // Initialize dispatch queue for concurrent enumeration
+    if (options & NSEnumerationConcurrent) {
+        queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+        group = dispatch_group_create();
+    }
+
+    __block BOOL stop = NO;
+    for (id curObj in self) {
+        if (options & NSEnumerationConcurrent) {
+            dispatch_group_async(group, queue, ^() {
```

Comment:
The reference platform will stop enumerating with NSEnumerationConcurrent when stop is set to `YES` but because of the asynchronicity we can't be sure if it stops any others that are currently running (if there's even a way to do that).  Given that, I think we should go with the implementation that's consistent w/ NSArray